galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt

```diff
@@ -656,43 +687,118 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
-      when(getSearchOperator()) {
+    TODO()
+  }
+
+  /**
+   * Returns the search conditions for this field.
+   */
+  open fun getSearchCondition_(): Pair<Expression<String>.(t: String) -> Op<Boolean>, Any?>? {
+
+    if (isNull(block!!.activeRecord)) {
+      fun nullCondBuilder(body: (scope: Expression<String>) -> Op<Boolean>): (Expression<String>.(t: String) -> Op<Boolean>) {
+        return (fun Expression<String>.(t: String): Op<Boolean> = body(this))
+      }
+
+      return when (getSearchOperator()) {
         VConstants.SOP_EQ -> null
-        VConstants.SOP_NE -> "IS NOT NULL"
-        else -> "IS NULL"
+        VConstants.SOP_NE -> (nullCondBuilder {
+          IsNotNullOp(it)
+        } to null)
+        else -> (nullCondBuilder {
+          IsNullOp(it)
+        } to null)
       }
     } else {
-      var operator = VConstants.OPERATOR_NAMES[getSearchOperator()]
+
+      var operator: (ExpressionWithColumnType<String>.(t: String) -> Op<Boolean>)? = null
+      // Operator name gets one of these operators as string , and will be the default value of operator
+      //eq - (==) neq - (!=) less - (<) lessEq - (<=) greater - (>) greaterEq - (>=)
+      //operator name = "=", "<", ">", "<=", ">=", "<>"
+      val operatorName = VConstants.OPERATOR_NAMES[getSearchOperator()]
       var operand = getSql(block!!.activeRecord)
 
+      fun <T, S1 : T?, S2 : T?> operatorBuilder(body: (scope: ExpressionWithColumnType<in S1>) -> ComparisonOp)
+              : (ExpressionWithColumnType<in S1>.(Any) -> Op<Boolean>)? {
+        return (fun ExpressionWithColumnType<in S1>.(_: Any): Op<Boolean> = body(this)) //as (ExpressionWithColumnType<String>.(Any) -> Op<Boolean>)?
+      }
+
+      fun <T, S : T?> ExpressionWithColumnType<in S>.wrap(value: T): QueryParameter<T> = when (value) {
+        is Boolean -> booleanParam(value)
+        is Byte -> byteParam(value)
+        is UByte -> ubyteParam(value)
+        is Short -> shortParam(value)
+        is UShort -> ushortParam(value)
+        is Int -> intParam(value)
+        is UInt -> uintParam(value)
+        is Long -> longParam(value)
+        is ULong -> ulongParam(value)
+        is Float -> floatParam(value)
+        is Double -> doubleParam(value)
+        is String -> QueryParameter(value, columnType) // String value should inherit from column
+        else -> QueryParameter(value, columnType)
+      } as QueryParameter<T>
+
+      operator = when (operatorName) {
+        "=" -> operatorBuilder {
+          EqOp(it, it.wrap(operand))
+        }
+        "<" -> operatorBuilder {
+          LessOp(it, it.wrap(operand))
+        }
+        ">" -> operatorBuilder {
+          GreaterOp(it, it.wrap(operand))
+        }
+        "<=" -> operatorBuilder {
+          LessEqOp(it, it.wrap(operand))
+        }
+        ">=" -> operatorBuilder {
+          GreaterEqOp(it, it.wrap(operand))
+        }
+        "<>" -> operatorBuilder {
+          NeqOp(it, it.wrap(operand))
```

Comment:
Here your operator receives a function that contains the operand and then at the end of the function you return a pair of operator & operand?? Check it!